src/cpp/session/modules/SessionPythonEnvironments.R

```diff
@@ -101,6 +101,27 @@
    if (file.exists(prefsPython))
       return(path.expand(prefsPython))
    
+   # check for Python on the PATH -- if we have an appropriate version of
+   # python3 or python on the PATH, then use it. we ignore python installations
+   # in /usr/bin because we primarily want to target python installations that
+   # appear to have been "intentionally" installed by the user. in addition,
+   # many Linux package managers install "minimized" versions of Python onto
+   # the system path that will fail to work without extra supporting packages
+   # installed, and managing and communicating all that state to the user is
+   # challenging
+   python3 <- Sys.which("python3")
+   if (nzchar(python3) && python3 != "/usr/bin/python3")
+      return(python3)
+   
+   # use a plain 'python' executable if it's not python 2
+   python <- Sys.which("python")
```

Comment:
In quarto, I fail this check on Windows unless it's conda (I think this is to avoid taking a version of Python on the PATH that is nevertheless not the preferred version of the `py` launcher). In other words, the user might have an older version of Python 3 on the PATH, then install Python 3.9 w/ the defaults (NOT placing Python 3.9 on the PATH) but we still don't get the newer version).

Reply:
On Windows, I also check for a definition of `PY_PYTHON`, and if it is defined use that in preference to everything.

Reply:
If I understand correctly, I think this means:

- We should check `PY_PYTHON` at the top since if it's set it should override any other auto-discovery logic,
- We should perform the `py` check on Windows first before checking the PATH.

Reply:
Yes, for Quarto if you set the `PY_PYTHON` environment variable that does indeed trump everything.